gnd/src/main/java/com/google/android/gnd/ui/offlinearea/OfflineAreaListAdapter.java

```diff
@@ -75,9 +90,42 @@ public void onClick(View v) {
   }
 
   @Override
-  public void onBindViewHolder(ViewHolder viewHolder, int position) {
-    viewHolder.binding.offlineAreaName.setText(offlineAreas.get(position).getId());
-    viewHolder.position = position;
+  public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
+    String areaName =
+        viewHolder.binding.getRoot().getContext().getString(R.string.offline_areas_unknown_area);
+
+    OfflineArea area = offlineAreas.get(position);
+    String id = area.getId();
```

Comment:
In particular, this is business logic that would be best defined in a more tightly scoped class imo.

Reply:
81c0b3f